umbral-pre-python/umbral_pre/__init__.pyi

```diff
@@ -123,14 +126,15 @@ def decrypt_original(delegating_sk: SecretKey, capsule: Capsule, ciphertext: byt
     ...
 
 
+@final
 class KeyFrag:
 
     def verify(
             self,
             verifying_pk: PublicKey,
-            delegating_pk: Optional[PublicKey],
-            receiving_pk: Optional[PublicKey],
-            ) -> VerifiedKeyFrag:
+            delegating_pk: Optional[PublicKey] = ...,
```

Comment:
Interesting. What does the `...` do?

Reply:
I don't think it should be there. Since it's a method from the Rust extension, it doesn't have any parameters with defaults (and `...` is not the correct type anyway, if anything it should be `None`)

Reply:
`Ellipsis`, or `...`, is a placeholder for an unwritten code. I used it because I ran into some issues with `mypy.stubtest` - It was convinced that the runtime type generated by `pyo3` was `Ellipsis`.

I looked into it again today and I found that the signature expansions are not working correctly because `pyo3` was unable to interpret types for `Option<&T>`. I manually overrode the signature to satisfy `mypy.stubtest` checks.

What initially added to my confusion were some left-over generated files between bindings compilation which made it harder to iterate. I fixed this by running `cargo clean && pip uninstall -y umbral_pre && git clean -xdf && pip install -e . && python -m mypy.stubtest umbral_pre`.